src/IssueViz/Editor/IIssueSpanCalculator.cs

```diff
@@ -97,21 +89,28 @@ internal IssueSpanCalculator(IChecksumCalculator checksumCalculator)
 
             var start = new SnapshotPoint(currentSnapshot, startPos);
             var end = new SnapshotPoint(currentSnapshot, endPos);
+            var snapshotSpan = new SnapshotSpan(start, end);
+
+            if (IsHashDifferent(range.LineHash, snapshotSpan.GetText()) &&
+                IsHashDifferent(range.LineHash, startLine.GetText()))
+            {
+                // Out of sync: the range reported in the diagnostic has been edited, so we can no longer calculate the span
+                return EmptySpan;
+            }
 
-            return new SnapshotSpan(start, end);
+            return snapshotSpan;
         }
 
-        private bool IsLineHashDifferent(ITextRange range, ITextSnapshotLine snapshotLine)
+        private bool IsHashDifferent(string issueHash, string documentText)
         {
-            if (string.IsNullOrEmpty(range.LineHash))
+            if (string.IsNullOrEmpty(issueHash))
```

Comment:
this check gets called 2 times + looks kind of out of place => remove it out of this method and place it before the main condition `if (IsHashDifferent(....`